### ISSUE TITLE
Ensure step 3 of the OverrideMimeType method for XHR matches the spec

### ISSUE THREAD

**Issue by KiChjang** [servo/servo#9551]
Code: `components/script/dom/xmlhttprequest.rs`
Spec: https://xhr.spec.whatwg.org/#the-overridemimetype%28%29-method

**Comment by apopiak** [comment]
I'd like to take this one.
Anything to look out for specifically?

**Comment by jdm** [comment]
```
14:08 <jdm> apopiak: pretty sure the bit about "excluding any parameters" is missing in the implementation, for one thing
14:11 <jdm> apopiak: it may also not convert to lowercase for things like FOO/BAR
```

**Comment by krstoff** [comment]
There hasn't been any activity on this. May I take over?

**Comment by apopiak** [comment]
Yes, I've had issues with the tooling and then it dropped off my awareness.

**Comment by jdm** [comment]
Go ahead!

**Comment by krstoff** [comment]
Okay, the code required was only a few lines, and I've got a test case written, but I'm a little bit confused by the way testing is done in Servo. This was a partially implemented feature so test cases for this request method already exist, so does that mean I need to change some expectations? Can someone give me a very brief walkthrough?

**Comment by jdm** [comment]
Yes, the [web-platform-tests](https://github.com/w3c/web-platform-tests/) testsuite (which we have a local clone for) is shared between browsers and supports running with expected output that can include fail, crash, timeout, etc. You can learn about updating expectations from the [readme](http://mxr.mozilla.org/servo/source/tests/wpt/README.md#80) for our test harness. Please ask questions if anything remains unclear :)

**Comment by jdm** [comment]
The easiest way to check is to run `./mach test-wpt tests/wpt/web-platform-tests/XMLHttpRequest/` and see if any tests now report different results. It may make sense to add your test as well, if it doesn't duplicate an existing one!

**Comment by KiChjang** [comment]
@krstoff Are you still working on this?

**Comment by jdm** [comment]
This is available for claiming!

**Comment by jfhumann** [comment]
I would like to work on this :)

**Comment by jdm** [comment]
Please do!

**Comment by KiChjang** [comment]
@jfhumann Is this still being worked on?

**Comment by jfhumann** [comment]
Yup it is :)

**Comment by KiChjang** [comment]
@jfhumann Is there a problem while working on this? I don't think it should take this long to make progress.

**Comment by KiChjang** [comment]
Unassigning since I didn't see progress for 20 days.

**Comment by bd339** [comment]
I would like to attempt this.

**Comment by jdm** [comment]
@bd339 Go for it!

**Comment by Wafflespeanut** [comment]
fixed by #10838
